### PR TITLE
TP-425: Updated to GigaSpaces 14.5 and Spring 5.1.7

### PR DESCRIPTION
**Pull request by kbs4j**
This PR updates astrix to `GS 14.5` and `Spring 5.1.7`
Notes to reviewer:
* #83 should be merged and conflicts resolved before this PR can be merged. 
* `gs-test` version needs to be updated once https://github.com/AvanzaBank/gs-test/pull/21 has been merged, built and released, before this PR can be merged.
* This PR should preferably be squash upon merge.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -125,24 +125,25 @@
...
+		<mockito.version>2.23.0</mockito.version>
 		<lifecycle-mapping.version>1.0.0</lifecycle-mapping.version>
 		<dropwizard.version>3.1.2</dropwizard.version>
-		<netty.version>4.0.31.Final</netty.version>
-		<gs-test.version>14.0.8</gs-test.version>
+		<netty.version>4.1.30.Final</netty.version>
 		<mimer-config.version>0.0.4</mimer-config.version>
 		<hystrix-multiconfig.version>0.0.3</hystrix-multiconfig.version>
+		<gs-test.groupId>com.avanza.gs14</gs-test.groupId>
+		<gs-test.version>14.5.0-SNAPSHOT</gs-test.version>
```
Comment (kbs4j) [inline pom.xml:146]: This version needs to be updated once built and released.

File: pom.xml [hunk 2]
```diff
@@ -457,27 +443,22 @@
...
-				<plugin>
-					<groupId>org.apache.maven.plugins</groupId>
-					<artifactId>maven-war-plugin</artifactId>
-					<version>2.1.azapatch</version>
-				</plugin>
 				<plugin>
 					<groupId>org.codehaus.mojo</groupId>
 					<artifactId>versions-maven-plugin</artifactId>
-					<version>2.8.1</version>
+					<version>2.5</version>
```
Comment (oskarwiksten) [inline pom.xml:461]: This change might have been accidental because of the previous cherry-picks.

Comment (kbs4j) [reply pom.xml:461]: Yes, Im working now on a overall clean-up and restructuring of the plugins. new commits are coming

File: pom.xml [hunk 3]
```diff
@@ -125,24 +125,50 @@
...
+		<maven-shade-plugin.version>2.4.1</maven-shade-plugin.version>
+		<maven-assembly-plugin.version>2.2.1</maven-assembly-plugin.version>
+		<maven-clean-plugin.version>2.5</maven-clean-plugin.version>
+		<maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
+		<maven-dependency-plugin.version>2.8</maven-dependency-plugin.version>
+		<maven-deploy-plugin.version>2.8.1</maven-deploy-plugin.version>
+		<maven-install-plugin.version>2.3.1</maven-install-plugin.version>
+		<maven-jar-plugin.version>2.3.1</maven-jar-plugin.version>
+		<maven-javadoc-plugin.version>3.2.0</maven-javadoc-plugin.version>
+		<maven-rar-plugin.version>2.2</maven-rar-plugin.version>
```
Comment (oskarwiksten) [inline pom.xml:158]: Maybe this is another plugin that we could remove?

Comment (kbs4j) [reply pom.xml:158]: Good idea! @oskarwiksten I will remove it and check the other plugins and verisons while I'm at it

File: pom.xml [hunk 4]
```diff
@@ -25,12 +25,12 @@
 		<module>doc-snippets</module>
 		<module>tutorial</module>
 		<module>astrix-all</module>
+		<module>astrix-bom</module>
 		<module>astrix-modules</module>
 		<module>astrix-metrics</module>
 		<module>astrix-contracts</module>
 		<module>astrix-netty-remoting</module>
 		<module>astrix-test-support</module>
-		<module>hystrix-multiconfig</module>
```
Comment (oskarwiksten) [inline pom.xml:33]: The contents of `hystrix-multiconfig` has been removed previously, and replaced with a with `<relocation>`. If  `hystrix-multiconfig` is removed as a module in this branch, then we should also consider removing `hystrix-multiconfig/pom.xml`.

Comment (kbs4j) [reply pom.xml:33]: Great point, removing it!

**Comment by kbs4j** [comment]
new commits added @oskarwiksten

File: pom.xml (hunk 1 above)
Comment (kbs4j) [reply pom.xml]: This is fixed

**Pull request merged**